kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMSettingsConverter.java

```diff
@@ -291,14 +405,17 @@ public static Map<String, Variant<?>> build80211WirelessSecuritySettings(Network
         KuraWifiSecurityType securityType = KuraWifiSecurityType.fromString(
                 props.get(String.class, KURA_PROPS_KEY_WIFI_SECURITY_TYPE, deviceId, propMode.toLowerCase()));
 
-        if (securityType == KuraWifiSecurityType.SECURITY_WEP) {
+        switch (securityType) {
+        case SECURITY_WEP:
             return createWEPSettings(props, deviceId, propMode);
-        } else if (securityType == KuraWifiSecurityType.SECURITY_WPA
-                || securityType == KuraWifiSecurityType.SECURITY_WPA2
-                || securityType == KuraWifiSecurityType.SECURITY_WPA_WPA2) {
+        case SECURITY_WPA:
+        case SECURITY_WPA2:
+        case SECURITY_WPA_WPA2:
             return createWPAWPA2Settings(props, deviceId, propMode);
-        } else {
-            throw new IllegalArgumentException("Security type \"" + securityType + "\" is not supported.");
+        case SECURITY_WPA2_WPA3_ENTERPRISE:
+            return createWPA2WPA3EnterpriseSettings(props, deviceId, propMode);
```

Comment:
Since you don't use any of the parameters you pass to the method I would propose to make Sonar happy and remove them.